Increment4-8

ncyellow opened the pull request.

ncyellow committed:
Рабочий вариант  чтения метрик через json

ncyellow committed:
Сделал изменение по роутеру /update/

ncyellow committed:
Заголов ответа  "Content-Type" = "application/json"

ncyellow committed:
Если нигде не накасячил то отправка работает по новому протоколу

ncyellow committed:
поправил опечатку

ncyellow committed:
гепотеза почему ломается тест

ncyellow committed:
закрою все проверки для автотестов

ncyellow committed:
Вернул провери на Content-Type добавил минимальный набор тестов. Нужно будет еще

ncyellow committed:
Реализовано задание пятого инкремента

ncyellow committed:
Пальцем в небо почему падают тесты.

ncyellow committed:
Рабочий вариант инеремента 6 кроме синхронного сброса. По нему вопросы

ncyellow committed:
Запуск тестов

ncyellow committed:
Поправил замечания стана

ncyellow committed:
Исправлена ошибка демаршалинга + тесты на сериализацию  и сохранение

ncyellow committed:
Исправлена ошибка изза которой при сериализации всегда всем проставлялось последнее значение

ncyellow committed:
Добавил парсинг флагов. Приоритет - переменные окружения важнее флагов

ncyellow committed:
Добавил middleware для сжатия gzip

ncyellow committed:
Для интереса попробую включить стандартный middleware от chi

ncyellow committed:
Переименовал по стану

ncyellow committed:
тест чекает Content-Type потому попровляю

ncyellow committed:
Раз тесты проходят пробую вернуть стандартный конвертер chi

ncyellow committed:
Небольшая реорганизация

ncyellow committed:
Мелкие правки по коду

ncyellow committed:
стан

ncyellow committed:
переделал тесты хендлеров на suite чтобы разделить тестирование хендлеров

ASPecherkin commented on internal/agent/agent.go:
```diff
@@ -133,12 +112,54 @@ func prepareGauges(metrics *Metrics) map[string]float64 {
+
+	result := make([]storage.Metrics, 0, len(gauges))
+	for name, value := range gauges {
+		// Если пользоваться value, то все значения будут ссылаться на одну и ту же переменную - последнюю
```
да, всё верно - хорошо, что отметил такую особенность - на неё многие натывкаются и не понимают почему это так 👍

ncyellow replied:
на самом деле наткнулся и я. но визуально было сразу понятно что дело в этом

ASPecherkin commented on internal/server/server.go:
```diff
@@ -0,0 +1,39 @@
+)
+
+type Server struct {
+	Conf config.Config
```
тут скорее стилистический коммент - обычно конфиг это ссылка на один объект на всё приложение 

почему так делают - если будет добавляться перечитывание конфигов в работающием приложении - ничего менять не придётся

ncyellow replied:
Заменил на указатель на конфиг и в агенте и в сервере

ASPecherkin commented on internal/server/storage/memory.go:
```diff
@@ -40,6 +41,15 @@ func (s *MapRepository) UpdateCounter(name string, value int64) error {
 }
 
+func (s *MapRepository) UpdateMetric(metric Metrics) error {
+	if metric.MType == Gauge {
```
я бы тут использовал switch - потому что типы метрик могут добавляться и с ним удобнее работать и читать

ncyellow replied:
ok

ASPecherkin commented on internal/server/storage/memory.go:
```diff
@@ -55,6 +65,34 @@ func (s *MapRepository) Counter(name string) (val int64, ok bool) {
 }
 
+func (s *MapRepository) Metric(name string, mType string) (val Metrics, ok bool) {
+	if mType == Gauge {
```
здесь тоже можно switch сделать

ASPecherkin reviewed (changes requested):
критичных замечаний нет, большинство просто мелкие стилистические замечания

хорошая работа

ncyellow committed:
Поправил замечания по ревью

1. Config это указатель и в агенте и в сервере
2. Заменены if на switch по замечаниям
